spec-0008/index.md

```diff
@@ -0,0 +1,227 @@
+Securely building release artifacts will be covered in a later SPEC. This set of recommendations complements the recommendations from [SPEC 6 — Keys to the Castle](https://github.com/scientific-python/specs/blob/main/spec-0006/index.md).
+
+This SPEC is written with GitHub in mind, note these recommendations still apply with other services such as GitLab and we refer project to these services' documentations and guidelines.
+
+## Implementation
+
+With a focus on securing the release artifact distribution process, the following processes and standards should be adopted.
+
+### Document the release process
+
+The release process should be clearly and fully documented in the developer documentation and describe each step to make a release and the permissions required to do so.
+
+### Hardening workflow environment permissions
+
+Workflows that publish release artifacts should have _run triggers_ that require intentional actions by maintainers (e.g., `on workflow_dispatch` in GitHub Actions) and require multiple maintainers to approve the workflow to run (c.f. "Use GitHub Actions environments" section below).
+This is to safeguard the project from any one maintainer having the ability to commit to the default branch and make a release directly.
+
+It is also strongly recommended that the repository requires signed commits so that all releases have a verified commit to which they correspond.
+The branch from which the release is made should also be protected.
+
+#### Restrict permissions in CI runners to the minimum required
+
+To restrict the attack surface area of arbitrary code execution in CI runners, the default permissions the runners have should be restricted to the minimum possible (read access). In the GitHub Action workflow, this is accomplished by defining the following workflow global permissions block before any jobs are defined.
+
+```yaml
+permissions:
+  content: read
+```
+
+Elevating permissions beyond this should be done at the job level by redefining the permissions block in the job.
+
+#### Restrict permitted actions in workflows
+
+GitHub allows to restrict the actions that workflows can use via repositories settings at `https://github.com/ORG/PROJECT/settings/actions`.
+A reasonable default setup is to:
+
+- allow actions from the repository itself and actions created by GitHub,
+- as well as an allow list of actions from third party repositories.
+
+  ```
+  OWNER/REPOSITORY@*
+  ```
+
+Consult [Managing GitHub Actions permissions for your repository](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#managing-github-actions-permissions-for-your-repository) for more details.
+
+#### Use GitHub Actions environments
+
+Use a GitHub Actions environment
```

Comment:
Maybe add a sentence or two about what GitHub Actions environments are, what they are good for, where to find more information, etc

For the snippet below write a sentence that explains what the snippet is, where it should go, link to GitHub docs on environments

Reply:
```suggestion
Use a [GitHub Actions environment](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment)
```